### PR TITLE
Example Health URL scheme with GET, single POST and batch POSTs

### LINKED ISSUE #154: Semantics for POST method
As part of our work on URL schemes and recommended common implementations, this issue covers using HTTP POST to send new resources from a client to a service. This issue is defined to allow discussion. I've made a start below:

## Goals
- Provide a standard approach that can be implemented by any server that allows clients to add new resources
- Work for any object collection where the server supports addition (for instance, animals, animal sets, events)
- Align with best practices for HTTP, REST, and JSON APIs
- Allow the posting of single resources and collections of resources
- Provides a standardised way of handling errors.

There are several parts to this issue:
1. How to determine if a collection implemented on the server supports adding resources
2. The method and URL path to use when adding new resources
3. Adding single resources or arrays/collections of resources
4. How to uniquely identify resources being added to the server
5. Determining status - success or errors
6. Return results when resources are successfully added
7. Return results when the server is processing new resources asynchronously
8. Returning error results.

## How to determine if a collection on a server supports adding resources
There are three options:
- Read the human-readable documentation or talk to an administrator
- Retrieve an icarResourceCollectionReference that provides a reference to the collection, and parse the operations array to see if the "POST" method is supported - this could be through a capabilities or endpoint enumerating API.
- Have some other standard capabilities API

## The method and URL path to use when adding new resources
1. The method should be HTTP POST as per [RFC-2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.5)
2. The URL should be the URL which one would GET to get a collection of resources of the specified type. For instance, the URL of the animals collection for icarAnimalCoreResource objects.

## Adding single resources or arrays/collections of resources
There might be several options:
- Only support posting individual items (likely inefficient for adding bulk data)
- Define a special purpose array/collection for posting
- Use an **icarResourceCollection**, setting the *memberType* and putting resources in the *members[]* array. The *view* properties would be unused.

## How to uniquely identify resources being added to the server
A discussion area in itself - broken out into issue #153 

## Determining status - success or errors
The recommended method for REST web services is to use [HTTP status codes ](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10).  For instance,
- **201** Created
- **202** Accepted
- **4xx** for client errors (for instance, **409** Conflict when a resource with the same unique ID already exists)
Substantial work on this has been done by JSON-API (https://jsonapi.org/format/#crud-creating-responses).

## Return results when resources are successfully added
Based on the RFC-2616 specifications and JSON-API, it would seem that when new resources are added successfully, the server should:
1. Return HTTP status 201 Created
2. Set the HTTP Location header to the location of the new resource(s)
3. Return the newly added resources (the same data sent, possibly with IDs filled in and other values generated) in the body.

 ## Return results when the server is processing new resources asynchronously
Based on the RFC-2616 specifications and JSON-API, it would seem that when the server is still processing the submitted, resources (and would otherwise time out), it should:
1. Return the HTTP status 202 Accepted
2. Return **something** in the body that indicates status and how to check status or the result. Some examples are given here: https://restfulapi.net/http-status-202-accepted/ (we will need to decided what we want to do).

##  Returning error results
The usual recommended approach is to return an HTTP 4xx client error status, and then return an error or problem response in the body.
- [RFC-7807](https://tools.ietf.org/html/rfc7807) defines a form of problem response that can be extended.
- JSON-API defines an [Errors object](https://jsonapi.org/format/#errors)
We will need to decide the representation we want to use AND whether a server should always fail when an error (rather than a warning) is encountered, or whether it should attempt to process other objects in the request.

### PR DESCRIPTION
**Pull request by cookeac**
Resolves #154 
* Adds healthURLScheme.json (OpenAPI example for animal health events)
* Adds icarBatchResult.json - results for batch POSTs
* Replaces icarErrorResource.json with icarErrorWarningMessageResource.json which is used for errors but also errors and warnings in batch results.
* Adds icarBatchResultSeverityType.json to provide an enumeration for error, warning, and informational messages.

In usage:
* GET methods function as they do in exampleURLScheme.json
* Single POST methods take a single resource and return the updated resource if successful, else an array of error/warnings.
* Batch POST methods take an array of resources and return a matching array of batch results, with the original metadata for matching, the updated resource IDs (where added successfully) and/or an array of errors and warnings for each resource.

### CODE REVIEW

File: resources/icarErrorWarningMessageResource.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  "description": "An RFC7807 compliant problem response for JSON APIs.",
```
Comment (ahokkonen) [inline resources/icarErrorWarningMessageResource.json]: Could "icarResponseMessageResource"/"icarProblemResponseMessageResource"  be a better name for that resource? As we also have "Information" severity level defined for message type.

**Review by ahokkonen — COMMENTED** [review]
Added one comment

File: resources/icarErrorWarningMessageResource.json (hunk 1 above)
Comment (cookeac) [reply resources/icarErrorWarningMessageResource.json]: Excellent suggestion, thank you. I struggled with the naming of that resource. Renamed to icarResponseMessageResource.

**Review by ahokkonen — APPROVED** [review]
Looks perfect now 👍🏻 Great work!

**Review by AndreasSchultzGEA — APPROVED** [review]
Hi, I had no closer look at the UrlScheme, but the other parts look fine.

**Pull request merged**